sweepai/handlers/on_ticket.py

```diff
@@ -201,15 +201,39 @@ def fetch_file_contents_with_retry():
     sweep_bot = SweepBot.from_system_message_content(
         human_message=human_message, repo=repo, is_reply=bool(comments)
     )
-    sweepbot_retries = 3
     try:
+        sweepbot_retries = 3
         for i in range(sweepbot_retries):
             logger.info("CoT retrieval...")
             if sweep_bot.model == "gpt-4-32k-0613":
                 sweep_bot.cot_retrieval()
             logger.info("Fetching files to modify/create...")
             file_change_requests = sweep_bot.get_files_to_change()
+            # Group file_change_requests by filename
```

Comment:
reduce the amount of lines changed, this is verbose for what we want to do. Also remove the 0 indexing where it's not appropriate